inspirehep/ext/jinja_filters/record.py

```diff
@@ -458,3 +427,105 @@ def is_upper(s):
     def split_author_name(name):
         new_name = name.split(',')
         return ' '.join(reversed(new_name))
+
+    @app.template_filter()
+    def publication_info(record):
+        """Displays inline publication and conference information"""
+        result = {}
+        out = []
+        """Publication info line"""
+        if 'publication_info' in record:
+            journal_title, journal_volume, year, journal_issue, pages = \
+                ('', '', '', '', '')
+            for pub_info in record['publication_info']:
+                if 'journal_title' in pub_info:
+                    journal_title = '<i>' + pub_info['journal_title'] + '</i>'
+                    if 'journal_volume' in pub_info:
+                        journal_volume = ' ' + pub_info['journal_volume']
+                    if 'year' in pub_info:
+                        year = ' (' + str(pub_info['year']) + ')'
+                    if 'journal_issue' in pub_info:
+                        journal_issue = ' ' + pub_info['journal_issue'] + ', '
+                    if 'page_artid' in pub_info:
+                        pages = ' ' + pub_info['page_artid']
+                    out.append(journal_title + journal_volume +
+                               year + journal_issue + pages)
+            if out:
+                result['pub_info'] = out
```

Comment:
Here you are returning a list. Might be better if you keep appending the parts of the string like:

```python
out.append(journal_title)
out.append(year)
```

And at the end:

```python
result['pub_info'] = "".join(out)
```

or

 ```python
result['pub_info'] = " ".join(out)
```

if you need each element separated by space.